src/elements/icon/icon.js

```diff
@@ -0,0 +1,42 @@
+(function() {
+  'use strict';
+
+  angular
+  .module('semantic.ui.elements.icon', [])
+  .directive('smIcon', smIcon);
+
+
+  function smIcon() {
+    return {
+      restrict: 'E',
+      replace: true,
+      transclude: true,
+      scope: {
+        fontIcon: '@smFontIcon',
+        iconSize: '@smIconSize'
+      },
+      template: getTemplate,
+      link: function(scope, element, attrs, ctrl, transclude) {
+        var node = element[0];
+
+        transclude(scope, function(nodes) {
+          element.append(nodes);
+        });
+
+        //Adding base icon class, if it is not mentioned explicitly.
+        scope.fontIcon = scope.fontIcon && scope.fontIcon.indexOf('icon') <= -1 ? scope.fontIcon + ' icon' : scope.fontIcon;
```

Comment:
Actually, this code doesn't work to begin with --- interpolated isolate scope variables will be overwritten with the original value almost immediately